bee-ledger/src/workers/snapshot/download.rs

```diff
@@ -154,13 +161,17 @@ async fn download_snapshot_file(path: &Path, download_url: &str) -> Result<(), E
 
     match reqwest::get(download_url).await {
         Ok(res) => {
-            tokio::io::copy(
-                &mut res.bytes().await.map_err(|_| Error::DownloadingFailed)?.as_ref(),
-                &mut tokio::fs::File::create(path).await?,
-            )
-            .await?;
+            if res.status().is_success() {
+                tokio::io::copy(
+                    &mut res.bytes().await.map_err(|_| Error::DownloadingFailed)?.as_ref(),
+                    &mut tokio::fs::File::create(path).await?,
+                )
+                .await?;
+            } else {
+                warn!("Downloading snapshot file failed with status code {:?}.", res.status());
+            }
         }
-        Err(e) => warn!("Downloading snapshot file failed with status code {:?}.", e.status()),
+        Err(e) => debug!("Downloading snapshot failed: {:?}.", e.to_string()),
```

Comment:
Do we download a file only if we managed to download the header ? If yes this should maybe stay as a warning ?

Reply:
You're right, and @Adam-Gleave was also right all along.

Reply:
But Adam's comment was on the header, so it should actually be debug there ?

Reply:
Yeah, the header should both be `debug` and the actual snapshot should be `warn`. This is probably what caused Adam to write the comment.

Reply:
Yes, the inconsistency with the errors was confusing me!